server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/issue/IssueLifecycle.java

```diff
@@ -65,21 +70,59 @@ public void initNewOpenIssue(DefaultIssue issue) {
     issue.setEffort(debtCalculator.calculate(issue));
   }
 
-  public void copyExistingOpenIssue(DefaultIssue raw, DefaultIssue base) {
+  public void copyExistingOpenIssueFromLongLivingBranch(DefaultIssue raw, DefaultIssue base, String fromLongBranchName) {
     raw.setKey(Uuids.create());
     raw.setNew(false);
-    raw.setCopied(true);
-    copyFields(raw, base);
+    copyIssueAttributes(raw, base);
+    raw.setFieldChange(changeContext, IssueFieldsSetter.FROM_LONG_BRANCH, fromLongBranchName, analysisMetadataHolder.getBranch().get().getName());
+  }
 
-    if (base.manualSeverity()) {
-      raw.setManualSeverity(true);
-      raw.setSeverity(base.severity());
+  public void mergeConfirmedOrResolvedFromShortLivingBranch(DefaultIssue raw, DefaultIssue base, String fromShortBranchName) {
+    copyIssueAttributes(raw, base);
+    raw.setFieldChange(changeContext, IssueFieldsSetter.FROM_SHORT_BRANCH, fromShortBranchName, analysisMetadataHolder.getBranch().get().getName());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)